app/models/assignment.rb

```diff
@@ -1156,6 +1156,30 @@ def to_json(options = {})
     self.assignment_properties.attributes.merge(self.attributes).symbolize_keys.to_json(options)
   end
 
+  # Returns an assignment's relevant properties for uploading/downloading an assignment's configuration as a hash
+  def assignment_properties_config
+    properties = {}
+    fields = DEFAULT_FIELDS + [:group_name_displayed, :repository_folder, :invalid_override,
+                               :section_groups_only, :section_due_dates_type, :unlimited_tokens,
+                               :only_required_files, :token_start_date, :token_period,
+                               :non_regenerating_tokens, :scanned_exam, :anonymize_groups,
+                               :hide_unassigned_criteria, :duration, :start_time, :is_timed,
+                               :starter_file_type, :starter_files_after_due, :type,
+                               :show_total, :outstanding_remark_request_count]
+    fields.each do |f|
+      properties[f] = self.assignment.send(f)
+    end
+    # Get all other assignment properties
+    properties[:submission_rule] = self.submission_rule.type
+    late_periods = self.submission_rule.periods.pluck_to_hash(:deduction, :hours, :interval)
+    section_dates = self.section_due_dates.pluck_to_hash(:due_date, :start_time)
+    required_files = self.assignment_files.pluck_to_hash(:filename)
+    # Merge all properties together
+    properties.merge('submission_rule_periods': late_periods)
```

Comment:
Don't chain merge like this. You can just pass multiple keys/value pairs as the arguments to a single merge call